hack/build-local-images.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/sh -e
+
+#imagenames=(docker-registry origin deployer recycler docker-builder                sti-builder                f5-router node openvswitch cluster-capacity federation)
+#imagedirs=( dockerregistry  origin deployer recycler builder/docker/docker-builder builder/docker/sti-builder router/f5 node openvswitch cluster-capacity federation)
+imagenames=(docker-registry origin deployer recycler docker-builder                sti-builder                f5-router node openvswitch)
+imagedirs=( dockerregistry  origin deployer recycler builder/docker/docker-builder builder/docker/sti-builder router/f5 node openvswitch)
+
+count=${#imagenames[@]}
+crc_count=${#imagedirs[@]}
+if [ $count -ne $crc_count ]; then
+    echo "imagenames and imagedirs arrays are different sizes, they need to be parallel arrays."
+    echo "$count $crc_count"
+    exit 1
+fi
+
+# reuse the same bin directory for all the docker builds
+# to save time copying stuff around.
+BIN_TMP=`mktemp -d`
+echo "Staging bin files in $BIN_TMP"
+#mkdir -p $BIN_TMP/bin
+#cp -a `pwd`/_output/local/bin/linux/amd64/* $BIN_TMP/bin
+ln -s `pwd`/_output/local/bin/linux/amd64 $BIN_TMP/bin
+
+pushd $BIN_TMP
+
+trap cleanup exit
+function cleanup {
+    popd
+    rm -rf $BIN_TMP    
+}
+
+c=-1
+while [ $c -lt $((count-1)) ]; do 
+    c=$((c+1))
+    imagename=${imagenames[$c]}
+    imagedir=${imagedirs[$c]}
+    tag_prefix="openshift/origin-"
+    if [[ $imagename == "origin" || $imagename == "node" || $imagename ==  "openvswitch" ]]; then
+        tag_prefix="openshift/"
+    fi
+
+
+    # for most images we are replacing the openshift binary, but for the
+    # docker registry image, we're replacing the dockerregistry binary.
+    # similar hacks for other images.
+    filename="openshift"
+    if [ $imagename == "docker-registry" ]; then
+        filename="dockerregistry"
+    fi
+
+    #if [ $imagename == "federation" ]; then
+    #    filename="hyperkube"
+    #fi
+
+    #if [ $imagename == "cluster-capacity" ]; then
+    #    filename="cluster-capacity"
+    #fi
```

Comment:
@derekwaynecarr i've commented out these two images until we sort out where the binaries they consume are actually coming from.